From ee276dfcc7d4b25214ec6745ebf55c4666b3bd0a Mon Sep 17 00:00:00 2001
From: Peter Xu <peterx@redhat.com>
Date: Wed, 19 Jun 2024 18:30:36 -0400
Subject: [PATCH 01/11] migration/multifd: Avoid the final FLUSH in complete()

RH-Author: Juraj Marcin <None>
RH-MergeRequest: 419: migration: New postcopy state, and some cleanups [rhel-9.5.z]
RH-Jira: RHEL-63874
RH-Acked-by: Peter Xu <peterx@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Commit: [1/11] 028e310f65eaad098ef62bdb8a5d30b9a5cd32e2

We always do the flush when finishing one round of scan, and during
complete() phase we should scan one more round making sure no dirty page
existed.  In that case we shouldn't need one explicit FLUSH at the end of
complete(), as when reaching there all pages should have been flushed.

Reviewed-by: Fabiano Rosas <farosas@suse.de>
Tested-by: Fabiano Rosas <farosas@suse.de>
Signed-off-by: Peter Xu <peterx@redhat.com>
Signed-off-by: Fabiano Rosas <farosas@suse.de>

(cherry picked from commit 637280aeb242517ede480aa2d5ba1c29d41eac11)

JIRA: https://issues.redhat.com/browse/RHEL-63874
Y-JIRA: https://issues.redhat.com/browse/RHEL-38485

Signed-off-by: Juraj Marcin <jmarcin@redhat.com>
---
 migration/ram.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/migration/ram.c b/migration/ram.c
index 8deb84984f..3ef84e7036 100644
--- a/migration/ram.c
+++ b/migration/ram.c
@@ -3383,10 +3383,6 @@ static int ram_save_complete(QEMUFile *f, void *opaque)
         }
     }
 
-    if (migrate_multifd() && !migrate_multifd_flush_after_each_section() &&
-        !migrate_mapped_ram()) {
-        qemu_put_be64(f, RAM_SAVE_FLAG_MULTIFD_FLUSH);
-    }
     qemu_put_be64(f, RAM_SAVE_FLAG_EOS);
     return qemu_fflush(f);
 }
-- 
2.39.3

